h2o-core/src/main/java/hex/ConfusionMatrix.java

```diff
@@ -23,50 +28,31 @@
    */
   public ConfusionMatrix(double[][] value, String[] domain) { _cm = value; _domain = domain; }
 
-  /** Build the CM data from the actuals and predictions, using the default
-   *  threshold.  Print to Log.info if the number of classes is below the
-   *  print_threshold.  Actuals might have extra levels not trained on (hence
-   *  never predicted).  Actuals with NAs are not scored, and their predictions
-   *  ignored. */
-  public static ConfusionMatrix buildCM(Vec actuals, Vec predictions) {
-    if (!actuals.isCategorical()) throw new IllegalArgumentException("actuals must be categorical.");
-    if (!predictions.isCategorical()) throw new IllegalArgumentException("predictions must be categorical.");
-    Scope.enter();
-    try {
-      Vec adapted = predictions.adaptTo(actuals.domain());
-      int len = actuals.domain().length;
-      CMBuilder cm = new CMBuilder(len).doAll(actuals, adapted);
-      return new ConfusionMatrix(cm._arr, actuals.domain());
-    } finally {
-      Scope.exit();
-    }
-  }
-
-  private static class CMBuilder extends MRTask<CMBuilder> {
-    final int _len;
-    double _arr[/*actuals*/][/*predicted*/];
-    CMBuilder(int len) { _len = len; }
-    @Override public void map( Chunk ca, Chunk cp ) {
-      if (_len > MAX_CM_CLASSES) return;
-      // After adapting frames, the Actuals have all the levels in the
-      // prediction results, plus any extras the model was never trained on.
-      // i.e., Actual levels are at least as big as the predicted levels.
-      _arr = new double[_len][_len];
-      for( int i=0; i < ca._len; i++ )
-        if( !ca.isNA(i) )
-          _arr[(int)ca.at8(i)][(int)cp.at8(i)]++;
-    }
-    @Override public void reduce( CMBuilder cm ) {
-      ArrayUtils.add(_arr,cm._arr);
-    }
-  }
-
-
   public void add(int i, int j) { _cm[i][j]++; }
 
   public final int size() { return _domain.length; }
 
-  boolean tooLarge() { return size() > MAX_CM_CLASSES; }
+  boolean tooLarge() { return size() > maxClasses(); }
+
+  static int maxClasses() {
+    String maxClassesSpec = System.getProperty(MAX_CM_CLASSES_KEY);
+    if (maxClassesSpec == null)
+      return MAX_CM_CLASSES_DEFAULT;
+    return parseMaxClasses(maxClassesSpec);
+  }
+  static int parseMaxClasses(String maxClassesSpec) {
+    try {
+      int maxClasses = Integer.parseInt(maxClassesSpec);
+      if (maxClasses <= 0) {
+        Log.warn("Using default limit of max classes in a confusion matrix (" + MAX_CM_CLASSES_DEFAULT + ", user specification is invalid: " + maxClasses + ")");
+        return MAX_CM_CLASSES_DEFAULT;
+      } else
+        return maxClasses;
+    } catch (NumberFormatException e) {
+      Log.err("Invalid specification of maximum classes of a confusion matrix: " + maxClassesSpec);
```

Comment:
Minor: On row 47, user is told default number of classes is used. Would be nice to inform the user here too.